reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java

```diff
@@ -2474,6 +2475,48 @@ void testConnectionClosePropagatedAsError_TE() throws Exception {
 				"POST http://%s/ HTTP/1.1\r\nHost: %s\r\nTransfer-Encoding: chunked\r\n\r\n10\r\nhello");
 	}
 
+	@Test
+	void testOrderRoutes() {
```

Comment:
we need to test that the comparator can be overridden and also removed so:

```
@Test
void testOrderRoutes() {
	HttpServerRoutes routes =
			HttpServerRoutes.newRoutes()
					.get("/yes/{value}", (request, response) -> response.status(200).send())
					.get("/yes/value", (request, response) -> response.status(201).send())
					.comparator(comparator);

	DisposableServer server = null;

	// 1. Case one .comparator(comparator)
	try {
		server = createServer().handle(routes).bindNow();

		StepVerifier.create(createClient(server.port()).get().uri("/yes/value")
				.response().map(httpClientResponse -> httpClientResponse.status().code()))
				.expectNext(201)
				.expectComplete()
				.verify(Duration.ofSeconds(30));
	}
	finally {
		if (server != null) {
			server.disposeNow();
		}
	}

	// 2. Case two .comparator(comparator.reversed())
	try {
		server = createServer().handle(routes.comparator(comparator.reversed())).bindNow();

		StepVerifier.create(createClient(server.port()).get().uri("/yes/value")
				.response().map(httpClientResponse -> httpClientResponse.status().code()))
				.expectNext(200)
				.expectComplete()
				.verify(Duration.ofSeconds(30));
	}
	finally {
		if (server != null) {
			server.disposeNow();
		}
	}

	// 3. Case three .noComparator()
	// TODO
}
```

Reply:
I add four test case：1、no comparator;2、set comparator；3、comparator can override(has one comparator)；4、clear comparator(set comparator and then clear)